servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/HeaderUtils.java

```diff
@@ -182,10 +183,12 @@ private static void updateResponseContentLength(final int contentLength, final H
                 flatRequest = from(metadata, EmptyHttpHeaders.INSTANCE);
             } else if (reduction instanceof List) {
                 final List<?> items = (List<?>) reduction;
-                for (int i = 0; i < items.size(); i++) {
-                    contentLength += calculateContentLength(items.get(i));
+                for (Object item : items) {
+                    contentLength += calculateContentLength(item);
                 }
-                flatRequest = Publisher.<Object>from(metadata).concat(fromIterable(items));
+                flatRequest = Publisher.<Object>from(metadata)
+                        .concat(fromIterable(items))
+                        .concat(succeeded(EmptyHttpHeaders.INSTANCE));
```

Comment:
Append trailers when the result of the reduction is a list of items.

Trailers should not be here, reasoning:
- When trailers exist according to spec, the transfer encoding should be chunked
- When the payload 'mayHaveTrailers==true' this flow is avoided
- The `calculateContentLength(item)` method already checks instance type and throws exception if not buffer.